helix-term/src/ui/completion.rs

```diff
@@ -79,7 +92,39 @@ impl menu::Item for CompletionItem {
                 Some(lsp::CompletionItemKind::ENUM) => "enum",
                 Some(lsp::CompletionItemKind::KEYWORD) => "keyword",
                 Some(lsp::CompletionItemKind::SNIPPET) => "snippet",
-                Some(lsp::CompletionItemKind::COLOR) => "color",
+                Some(lsp::CompletionItemKind::COLOR) => {
+                    let maybe_hex_color = match &item.documentation {
+                        Some(Documentation::String(text)) => Some(text),
+                        Some(Documentation::MarkupContent(MarkupContent { value, .. })) => {
+                            Some(value)
+                        }
+                        None => None,
+                    };
+
+                    let content = maybe_hex_color.map_or(Span::raw("color"), |hex_color| {
+                        let maybe_color = match (
+                            hex_color
+                                .get(1..=2)
+                                .and_then(|r| u8::from_str_radix(r, 16).ok()),
+                            hex_color
+                                .get(3..=4)
+                                .and_then(|g| u8::from_str_radix(g, 16).ok()),
+                            hex_color
+                                .get(5..=6)
+                                .and_then(|b| u8::from_str_radix(b, 16).ok()),
+                        ) {
+                            (Some(r), Some(g), Some(b)) => Some(Color::Rgb(r, g, b)),
+                            _ => None,
+                        };
+
+                        match maybe_color {
+                            Some(color) => Span::styled("        ", Style::default().bg(color)),
```

Comment:
For the display, I want to keep the word "color" even if we're showing a preview in the menu for accessibility reasons. I'm imagining `color ■` where the box has the foreground of the color. To do that, `kind` will need to be `Spans`:

```rust
Spans::from(vec![
    Span::raw("color "),
    Span::styled("■", Style::default().fg(color))
])
```

and that should work well with `.into()` - `.into()` should still cover all of the other branches of the match

Reply:
I did it in https://github.com/helix-editor/helix/pull/12299/commits/b911d0e6af24680a53f12bbed87615228754f2a9

In order to return `Spans` from the "color" branch i needed to return `Spans` from all branches

I thought that using `Span::from(vec!["constant"])` everywhere may be less performant so what I'm doing is returning a `Span` from each

then doing it like this:

```rs
            if kind.content == "■" {
                menu::Cell::from(Spans::from(vec![Span::raw("color "), kind]))
            } else {
                menu::Cell::from(kind)
            },
```